### DISCUSSION TITLE
Multiple nnn instances open

### DISCUSSION THREAD

**Discussion by mizlan** [mcchrish/nnn.vim#102] [Q&A] [answered]
Does this plugin support multiple instances of nnn open at the same time? I'm currently working on a persistent file explorer feature and am foreseeing the possibility that someone may have a file-explorer nnn on the left-hand side while also opening up another floating nnn instance. This may cause problems.

Not to mention in general, users can create multiple tabpages and open multiple nnn instances.

**Comment by mcchrish** [comment, marked as answer]
I think it does not but I haven't tried completely.

https://github.com/mcchrish/nnn.vim/blob/abf700104c0450bac10be3ece613ff88997f47e4/autoload/nnn.vim#L1-L3

Since the temp_file used to store picked files and tbuf being a source variable in the file, I have to assume multiple instance is not supported. It's possible to move out these variables into local instances though but it requires some refactoring.

**Reply by mcchrish** [reply]
@mizlan I converted the `s:tbuf` to a buffer variable`b:tbuf` that is local to the terminal buffer. It seems to be working fine so far. It may help with adding support to mulitple instances as it's no longer tied to the source file.
